.github/workflows/pr-ci.yml

```diff
@@ -1,85 +1,169 @@
 name: CI build and push (PR)
 
 concurrency:
-  group: pr-${{ github.event.number }}
+  group: ci-${{ github.run_id }}
   cancel-in-progress: true
 
 on:
-  pull_request_target:
+  issue_comment:
     types:
-      - opened
-      - reopened
-      - synchronize
-      - labeled
-      - unlabeled
-    branches:
-      - main
-      - v[0-9]+
-      - v[0-9]+.[0-9]+
-      - cryostat-v[0-9]+.[0-9]+
+      - created
 
 jobs:
+  check-before-build:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'cryostatio' && github.event.issue.pull_request && startsWith(github.event.comment.body, '/build_test')
+    steps:
+    - name: Fail if needs-triage label applied 
+      if: ${{ contains(github.event.issue.labels.*.name, 'needs-triage') }}
+      run: exit 1
+    - name: Show warning if permission is denied
+      if: |
+        !(github.event.comment.author_association == 'MEMBER' ||
+        github.event.comment.author_association == 'OWNER')
+        && !contains(github.event.issue.labels.*.name, 'safe-to-test')
+      uses: thollander/actions-comment-pull-request@v2
+      with:
+        message: |-
+          You do not have permission to run the /build_test command. Please ask Cryostat @reviewers 
+          to resolve the issue.
+    - name: Fail if command permission is not allowed
+      if: |
+        !(github.event.comment.author_association == 'MEMBER' ||
+        github.event.comment.author_association == 'OWNER')
+        && !contains(github.event.issue.labels.*.name, 'safe-to-test')
+      run: exit 1
+    - name: Show warning if safe-to-test applied and permission denied
+      if: |
+        !(github.event.comment.author_association == 'MEMBER' ||
+        github.event.comment.author_association == 'OWNER' ||
+        github.event.issue.user.name == github.event.comment.user.name)
+      uses: thollander/actions-comment-pull-request@v2
+      with:
+        message: |-
+          You do not have permission to run the /build_test command. Please ask Cryostat @reviewers 
+          to resolve the issue.
+     - name: Fail if safe-to-test applied and command permission is not allowed
+      if: |
```

Comment:
I think u can just use this if for step `Fail if command permission is not allowed` and remove this.

Reply:
Btw, pls rename it to `Fail if command permission is denied` just for consistency.

Reply:
Something like should work?

```yaml
!(github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'OWNER')
&& 
(!contains(github.event.issue.labels.*.name, 'safe-to-test') || github.event.issue.user.name != github.event.comment.user.name))
```

This should translate to:

> Permission denied if the user is nether OWNER or MEMBER and either `safe-to-test` label is not applied or the user is not the author when the label is applied.